### PR TITLE
Non-Persistent HTTP RPC Support

### PR DESCRIPTION
**Pull request by cihangir**
Hi, we have a microservice system based on rpcplus, we now need to export most of our services to the browser, since rpc uses a persistent connection and connects with "CONNECT" method, i refactored some part of the code and added a "server" that handles single open-close request responses. It has so small differences from other "servers"

With this change anyone will be able to use rpcplus as their web api server.

This is an example for  http api

````
func main() {
	server := rpcplus.NewServer()
	server.Register(new(methods.Arith))

	mux := http.NewServeMux()

	contextCreator := func(req *http.Request) context.Context {
		return context.Background()
	}

	rpcwrap.ServeHTTPRPC(
		mux,                    // httpmuxer
		server,                 // rpcserver
		"json",                 // codec name
		jsonrpc.NewServerCodec, // jsoncodec
		contextCreator,         // contextCreator
	)

	fmt.Println("Server listening on 3000")
	http.ListenAndServe("localhost:3000", mux)
}
```

### CODE REVIEW

**Comment by henryanand** [comment]
The NewRelic license key is a secure environment variable in Travis, so it is unset for forked repo pull requests.

http://docs.travis-ci.com/user/environment-variables/#Secure-Variables

The dependency script has been modified to skip NewRelic if the license key is unset. Please merge master and retry.

https://github.com/youtube/vitess/pull/281

File: go/rpcwrap/rpcwrap.go [hunk 1]
```diff
@@ -185,3 +185,75 @@ func GetRpcPath(codecName string, auth bool) string {
 	}
 	return path
 }
+
+// ServeCustomRPC serves the given http rpc requests with the provided ServeMux,
```
Comment (sougou) [inline go/rpcwrap/rpcwrap.go]: ServeCustomRPC -> ServeHTTPRPC.
Can you run golint and see if there are other new warnings?

Comment (cihangir) [inline go/rpcwrap/rpcwrap.go]: Sure, will fix them later today

**Comment by cihangir** [comment]
@sougou  I updated the pr with the golint fixes, PTAL.

@henryanand  thanks! At least unit tests are passing now:) others are still no go... this time zk and mysql

**Comment by sougou** [comment]
LGTM
Thanks for going above and beyond. I was only expecting you to fix lint errors in your diff.
I'll merge this once @henryanand takes a look at the travis errors.

**Comment by sougou** [comment]
@henryanand is this ok to merge, or do you wan to try and fix the build?

**Comment by henryanand** [comment]
Sorry for the delayed response, as I was OOO. These integration tests were fixed in https://github.com/youtube/vitess/pull/286 Please merge master and retry.

**Pull request merged**